src/allmydata/interfaces.py

```diff
@@ -2936,3 +2936,68 @@ class IConnectionStatus(Interface):
         (pending, connected, refused, or other errors).
         """)
 
+
+class ITahoeUsagePolicy(Interface):
+    """
+    Callbacks in this interface are called by core Tahoe-LAFS APIs,
+    thus controlling whether particular API calls are allowed or not.
+
+    There should be enough information and hook points to support at
+    least these use-cases:
+
+      - anonymous, everything allowed (current default)
+      - "storage club" / "friend-net" (possibly identity based?)
+      - cryptocurrencies (ideally, paying for each API call?)
+      - anonymous tokens (payment for service, but without identities)
+    """
+
+    @property
+    def name():
+        """
+        How this policy shall be referred to (e.g. in Storage Server
+        announcements, to users, etc). Ideally limit yourself to
+        lowercase letters, numbers (but not a leading number) and
+        underscores.
+        """
+
+    # XXX what sort of information should we pass in here? the "Node"
+    # (subclass) instance perhaps (should be able to get to everything
+    # possible from there?) as a worst case (worst case, because then
+    # plugin authors are exposed to the core of the beast)...but it
+    # needs at least the config (but probably some other things?) if
```

Comment:
Perhaps we can supply *just* the config that belongs to this plugin?

If we have something like:

```
[storage]
authorizers=
    https://tahoe-lafs.org/authz/anonymous/v1

[authorizer.https://tahoe-lafs.org/authz/anonymous/v1]
max_frobs = 12
```

then we can pass `{"max_frobs": "12"}` to whatever the plugin initializer is.

If there's a less abusive way to do this with ini files, that's fine with me too.